From 2da159715f7a9a6d12146a475d44e30cd57713da Mon Sep 17 00:00:00 2001
From: Anju Kunjumon <akunjumon@sierrawireless.com>
Date: Tue, 7 Sep 2021 16:09:13 -0700
Subject: [PATCH] conf: Allow only approved Cipher suites

Ensure only those ciphersuites from Sierra approved list are enabled by
default on Sierra Wireless products. The approved list is given below:
- All TLS 1.3 cipher suites and key exchange algorithms are approved.
- For TLS 1.2 and earlier the following cipher suites are approved for
  use with D/TLS. They are presented in order of preference.
   TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256
   TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
   TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
   TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
   TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
   TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
   TLS_ECDHE_ECDSA_WITH_AES_256_CCM
   TLS_ECDHE_ECDSA_WITH_AES_128_CCM
   TLS_ECDHE_ECDSA_WITH_AES_256_CCM_8
   TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8
   TLS_ECDHE_PSK_WITH_CHACHA20_POLY1305_SHA256
   TLS_ECDHE_PSK_WITH_AES_256_GCM_SHA384
   TLS_ECDHE_PSK_WITH_AES_128_GCM_SHA256
   TLS_ECDHE_PSK_WITH_AES_128_CCM_SHA256
   TLS_ECDHE_PSK_WITH_AES_128_CCM_8_SHA256
   TLS_PSK_WITH_CHACHA20_POLY1305_SHA256
   TLS_PSK_WITH_AES_256_GCM_SHA384
   TLS_PSK_WITH_AES_128_GCM_SHA256
   TLS_PSK_WITH_AES_256_CCM
   TLS_PSK_WITH_AES_128_CCM
   TLS_PSK_WITH_AES_256_CCM_8
   TLS_PSK_WITH_AES_128_CCM_8

Use "CipherString" setting for TLS1.2 and "Ciphersuites" setting for
TLS1.3 in openssl.cnf to filter the algorithms to allow only those in
the approved list.

Signed-off-by: Anju Kunjumon <akunjumon@sierrawireless.com>
---
 apps/openssl.cnf | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/apps/openssl.cnf b/apps/openssl.cnf
index 4acca4b..30f63fd 100644
--- a/apps/openssl.cnf
+++ b/apps/openssl.cnf
@@ -11,6 +11,21 @@
 # defined.
 HOME			= .
 
+openssl_conf = default_conf
+
+[default_conf]
+ssl_conf = ssl_sect
+
+[ssl_sect]
+system_default = system_default_sect
+
+[system_default_sect]
+# All TLS 1.3 cipher suites and key exchange algorithms are approved. Nothing to filter.
+# Ciphersuites = 
+#
+# TLS1.2 approved list of Cipher suites
+CipherString = ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-CCM:ECDHE-ECDSA-AES128-CCM:ECDHE-ECDSA-AES256-CCM8:ECDHE-ECDSA-AES128-CCM8:ECDHE-PSK-CHACHA20-POLY1305:ECDHE-PSK-AES256-GCM-SHA384:ECDHE-PSK-AES128-GCM-SHA256:ECDHE-PSK-AES128-CCM-SHA256:ECDHE-PSK-AES128-CCM8-SHA256:PSK-CHACHA20-POLY1305:PSK-AES256-GCM-SHA384:PSK-AES128-GCM-SHA256:PSK-AES256-CCM:PSK-AES128-CCM:PSK-AES256-CCM8:PSK-AES128-CCM8
+
 # Extra OBJECT IDENTIFIER info:
 #oid_file		= $ENV::HOME/.oid
 oid_section		= new_oids
-- 
2.17.1

